### DISCUSSION TITLE
Changing text and icon of a CommandButtonProjection

### DISCUSSION THREAD

**Discussion by MichaelWDenney** [kirill-grouchnikov/radiance#481] [Q&A] [answered]
Is there no means of changing the text and icon of a `CommandButtonProjection` without rebuilding the entire object?  I have a projection in the anchored section for which I need to dynamically change.  Rebuilding the object works, but then I need to remove the old object as an anchored command and add the new one.

**Comment by kirill-grouchnikov** [comment]
Do `BaseCommand.setText()` and `BaseCommand.setIconFactory()` not work for you? They are wired to propagate the changes to all command buttons created from the command. If it's not working specifically for anchored section in the ribbon, that's a bug.

**Comment by MichaelWDenney** [comment]
Hmmm.  Well. `setText()` and `setIconFactory()` aren't available for a CommandButtonProjection directly.  If I try `CommandButtonProject.getContentModel().setText()`, I'm not seeing any change, while `CommandButtonProjection.getContentModel().setIconFactory()` removes the icon rather than update it.

**Comment by kirill-grouchnikov** [comment]
See https://github.com/kirill-grouchnikov/radiance/commit/4c3aaceee8bc1e5bbf9d6a579b8c126e252b3da3 for how updates to command text and icon automatically propagate to the anchored command button. There is a bug in there that the new icon is displayed too big, not respecting the command button state. This will be fixed.

**Comment by kirill-grouchnikov** [comment, marked as answer]
The icon sizing has been fixed. Since this is not working for you in your app, I'll need to see a reproduced either in the `BasicCheckRibbon` demo, or in a small standalone sample.

**Reply by MichaelWDenney** [reply]
My mistake.  I don't know what happened earlier, but I just tried again and `CommandButtonProject.getContentModel().setText()` *is* working.  Sorry about that.  Thanks for fixing the icon sizing.

**Reply by kirill-grouchnikov** [reply]
Thanks for confirming, closing this one.

**Comment by kirill-grouchnikov** [comment]
(Really my first experience with GitHub discussions) - is it better to keep them open so that they can be more easily browsed?

**Reply by MichaelWDenney** [reply]
To be honest, I'm not very experienced with GitHub discussions either.  😄  I noticed that the top-level discussions page defaults to a search of `is:open` so discussions will be missed if someone is just browsing and doesn't remove that.  But if they are searching for something specific, they will override that search anyway.

**Reply by kirill-grouchnikov** [reply]
Let's keep it open for now. You marked it as answered, and that label appears in a nice green bold treatment in the list, so that's good.
